### DISCUSSION TITLE
Dynamically setting Cluster title property

### DISCUSSION THREAD

**Discussion by Slangenmans** [filamentphp/filament#12039] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

I'm trying to apply a translation to the title of a Cluster (see highlighted text in image). Using the title property will edit the text, however it needs to be set dynamically. I'm pretty sure getTitle() is the method I need, but I can't get it to work.  

```
class Categories extends Cluster
{
    protected static ?string $navigationIcon = 'heroicon-o-squares-2x2';

    public static function getNavigationLabel(): string
    {
        return __('filament_admin_panel.categories');
    }

    protected static ?string $title = 'Text to be edited';

    public function getTitle(): string|Htmlable
    {
        return 'this doesn\'t work';
    }
}
```

![image](https://github.com/filamentphp/filament/assets/68337699/2ec2746b-f95a-4ee3-8429-d63d086f7a6c)

**Comment by dissto** [comment]
Try 

```php
protected ?string $heading = 'Foo';
protected ?string $subheading = 'Bar';

// or 
public function getHeading(): ?string
{
    return 'Foo';
}

public function getSubheading(): string|Htmlable|null
{
    return 'Bar';
}
```
on the ListXXX class for each of your clusters items. For example your ListFaqs class.

**Reply by Slangenmans** [reply]
Thanks for the response! Unfortunately, the headings and subHeading properties do not change the highlighted text. I would have been very surprised if methods for properties relating to the items in the cluster edited the cluster's title. 

![image](https://github.com/filamentphp/filament/assets/68337699/fd0bce76-4fde-4b24-b0e9-2a079bb3b4f5)

**Reply by dissto** [reply, marked as answer]
I see. Then you probably want to take a look at the `getBreadcrumb` function or `$breadcrumb`

**Reply by Slangenmans** [reply]
`getClusterBreadcrumb` on the cluster class worked! Thank you very much for the quick responses :+1:
